### PR TITLE
Better binary data support in Expression/LG functions

### PR DESCRIPTION
**Pull request by Danieladu**
fixes: #5117

### Support binary input for prebuilt function `string`
Now, `string` would support converting a binary object into a normal string with `UTF-8` decoding.
For example:
`string(binary('hello'))`  -> 'hello'

### Refine `fromFile` 
Originally, `fromFile` accepts only one parameter that refers to the file path. The step is as follows:
- Find file
- Read file with string stream and get the string result
- Evaluate the text

In this PR, we introduce the second parameter of `fromFile`. 
Its value could be: `evaluated (default)`, `raw` or `binary`

So the change of this function is:
`fromFile(path: string)` ->
`fromFile(path: string, format: 'evaluated'(default) | 'raw' | 'binary')`

#### `raw` format
In some situations, users do not want to evaluate the text, and just want to keep the original content text.

Example:
There is a file, xx.txt, content is:
```
hi ${name}
```

Original, user could use `fromFile('xx.txt')` to get `hi jack`, `hi Lucy`,etc. according to the memory/properties passed in. 
But there is no way to get the original text. 
Now, user could use `fromFile('xx.txt', 'raw')` to get `hi ${name}`

#### `binary` format
In another situation, the user wants to get the binary result of the file.

Example:
`string(fromFile('xx.txt', 'binary'))`  => 'hi ${name}'
If you want to evaluate it directly, just append `expandText` function:
`expandText(string(fromFileBinary('xx.txt')))`  => 'hi Jack', 'hi Lucy' ...

Another example for Bot builder:
For Bot response, if the user wants to send a local image to the client, here is the solution:
```
# attachmentTemplate
[Attachment
    name = myImageName
    contentType = image/png
    contentUrl = data:image/png;base64,${${base64(fromFile(xx/name.png, 'binary'))}}
]
```

### CODE REVIEW

**Comment by boydc2014** [comment]
Better binary data support is good. 

And i'm not sure we need a new function, i thought we are extending from

`fromFile(path: string)`  =>
`fromFile(path: string, format: 'evaluated'(default) | 'raw' | 'binary')`

will that be more simpler and also be more explicit, if i'm asking raw, the function call will look like fromFile("a.txt", "raw"), rather than fromFile("a.txt", false)

**Comment by mrivera-ms** [comment]
@luhan2017, could you please review? Thanks.

File: tests/Microsoft.Bot.Builder.LanguageGeneration.Tests/TemplatesTest.cs [hunk 1]
```diff
@@ -1714,6 +1714,23 @@ public void TestInjectLGWithoutNamespace()
             Assert.Equal("hi Alice", evaled.ToString());
         }
 
+        [Fact]
+        public void TestFileOperation()
+        {
+            var templates = Templates.ParseFile(GetExampleFilePath("FileOperation.lg"));
+            var evaled = templates.Evaluate("FromFileWithoutEvaluation");
```
Comment (mrivera-ms) [inline tests/Microsoft.Bot.Builder.LanguageGeneration.Tests/TemplatesTest.cs:1721]: var name should be "evaluated".

**Review by mrivera-ms — CHANGES_REQUESTED** [review]

**Comment by boydc2014** [comment]
LGTM, just want to add some background, given the linked issue is too brief. So this is the issue we summarized from a few previous issues, which are related to loading\manipulating\sending files in adaptive dialog. In the process of supporting those issues, the piece we missing here is a better generic binary support. 

cc @mrivera-ms @luhan2017

**Comment by mrivera-ms** [comment]
@luhan2017, could you please review this PR? Thank you.

**Review by boydc2014 — APPROVED** [review]

**Pull request merged**